src/transforms.jl

```diff
@@ -1,3 +1,20 @@
+function transform_remove_local(ex)
+  ex isa Expr && ex.head === :local && return Expr(:block)
+  return ex
+end
+
+function transform_macro(ex)
+  ex isa Expr || return ex
+  ex.head !== :macrocall && return ex
+  return Expr(:call, :__secret__, ex.args)
```

Comment:
Is a global const `gensym` viable here instead of `__secret__`? I guess `gensym` is not deterministic so this can be bad. What about just a `struct MacroMarker end` instance?

I just want to avoid the exceedingly rare possibility that some other weird metaprogramming package is using the same symbol.